From 14770bd751d1004302016e53a0ea448ba075bc81 Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Fri, 22 May 2020 11:01:07 -0500
Subject: [PATCH 32/32] arm64: dts: beacon-imx8mm: correct ldo1/ldo2 voltage
 range

There is an up-stream patch [1] for the 8MM EVK that reads:
Correct ldo1 voltage range from wrong high group(3.0V~3.3V) to low group
(1.6V~1.9V) because the ldo1 should be 1.8V. Actually, two voltage groups
have been supported at bd718x7-regulator driver, hence, just corrrect the
voltage range to 1.6V~3.3V. For ldo2@0.8V, correct voltage range too.
Otherwise, ldo1 would be kept @3.0V and ldo2@0.9V which violate i.mx8mm
datasheet as the below warning log in kernel:

This should be applied to the Beacon i.MX8MM board as well.

[1] - https://lkml.org/lkml/2020/5/21/1205

Signed-off-by: Adam Ford <aford173@gmail.com>
---
 arch/arm64/boot/dts/freescale/beacon-imx8mm-som.dtsi | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/beacon-imx8mm-som.dtsi b/arch/arm64/boot/dts/freescale/beacon-imx8mm-som.dtsi
index 75cf1ef1e27b..012375d1249e 100644
--- a/arch/arm64/boot/dts/freescale/beacon-imx8mm-som.dtsi
+++ b/arch/arm64/boot/dts/freescale/beacon-imx8mm-som.dtsi
@@ -128,7 +128,7 @@
 
 			ldo1_reg: LDO1 {
 				regulator-name = "LDO1";
-				regulator-min-microvolt = <3000000>;
+				regulator-min-microvolt = <1600000>;
 				regulator-max-microvolt = <3300000>;
 				regulator-boot-on;
 				regulator-always-on;
@@ -136,7 +136,7 @@
 
 			ldo2_reg: LDO2 {
 				regulator-name = "LDO2";
-				regulator-min-microvolt = <900000>;
+				regulator-min-microvolt = <800000>;
 				regulator-max-microvolt = <900000>;
 				regulator-boot-on;
 				regulator-always-on;
-- 
2.17.1

